### ISSUE TITLE
Suggestion: Use Promise rather than callbacks

### ISSUE THREAD

**Issue by nkolban** [psiphi75/compass-hmc5883l#4]
It appears that JavaScript Promises have been available in Node since v4.0.0.  This issue is a feature request to provide a JavaScript Promises based interface to the functions rather than a callback interface.

**Comment by psiphi75** [comment]
I totally agree, I'd love to upgrade it to use promises and async/await.  However, I don't have a HMC5883l to test this on.  This library was written when the only version of Node.js that could be used on the BeagleBone (with Bonescript) was v0.12.x.  Hence the callbacks.

PR's are welcome though :).
